src/datumaro/components/algorithms/hash_key_inference/hashkey_util.py

```diff
@@ -172,3 +172,15 @@ def calculate_hamming(B1, B2):
     :return: hamming distance [r]
     """
     return np.count_nonzero(B1 != B2, axis=1)
+
+
+def match_query_subset(query_id, dataset):
+    subset_names = list(dataset.subsets().keys())
+    for subset_name in subset_names:
+        try:
+            query_datasetitem = dataset.get(query_id, subset_name)
+        except Exception:
+            continue
+        if query_datasetitem:
+            break
+    return query_datasetitem
```

Comment:
Please add `subset = None` argument to the function input signatures and CLI argument parser and iterate over subsets if `subset == None`. In addition, it can produce an empty list. It seems that we raise an appropriate error in that case.